### PR TITLE
Fix inline code formatting for text enclosed in backticks

### PR DESCRIPTION
**Pull request by marwan37**
## Please describe your changes

In the current implementation of the [@tiptap/extension-code](https://github.com/ueberdosis/tiptap/blob/39cf6979c49e953118bbbe4b894a1dc296128932/packages/extension-code/src/code.ts#L4), the regex pattern does not correctly format text enclosed in backticks when they are immediately preceded or followed by any character other than whitespace.

## How did you accomplish your changes

Updated the `inputRegex` and `pasteRegex` patterns to the following:

```ts
export const inputRegex = /(?<!`)`([^`]+)`(?!`)/;
export const pasteRegex = /(?<!`)`([^`]+)`(?!`)/g;
```

The regex uses negative lookbehind and lookahead assertions to ensure that only single backticks are considered for formatting to avoid conflicts with potential code block syntax or other uses of multiple backticks.

## How have you tested your changes

Tested in various scenarios to ensure that it correctly formats text as inline code blocks, and that it doesn't conflict with other extensions.

TipTap-LocalTest

## How can we verify your changes

1. Clone the repository and switch to the branch containing these changes.
2. Start the app via `npm run start` and navigate to any playground
3. Type text enclosed in backticks, with various surrounding characters.
4. You should observe the text within the backticks being correctly formatted as inline code.
5. Additionally, test with multiple backticks to ensure they don't interfere with the formatting.

## Remarks

This change enhances the flexibility of the inline code formatting.

## Checklist

- [x] The changes are not breaking the editor
- [x] Added tests where possible
- [x] Followed the guidelines
- [x] Fixed linting issues

## Related issues

#4467 
[Issue #338 in BlockNote](https://github.com/TypeCellOS/BlockNote/issues/338) - Original issue where the bug was identified and discussed.

### CODE REVIEW

File: packages/extension-code/src/code.ts [hunk 1]
```diff
@@ -34,14 +34,21 @@ declare module '@tiptap/core' {
...
+ * Regular expressions to match inline code blocks enclosed in backticks.
+ *  It matches:
+ *     - An opening backtick, followed by
+ *     - Any text that doesn't include a backtick (captured for marking), followed by
+ *     - A closing backtick.
+ *  This ensures that any text between backticks is formatted as code,
+ *  regardless of the surrounding characters (exception being another backtick).
  */
-export const inputRegex = /(?:^|\s)(`(?!\s+`)((?:[^`]+))`(?!\s+`))$/
+ */
```
Comment (nperez0111) [inline packages/extension-code/src/code.ts:45]:
```suggestion

```

**Review by nperez0111 — APPROVED** [review]
Thanks for your contribution @marwan37, sorry for the long wait!

**Comment by rfgamaral** [comment]
This change introduced the following issue:

firefox_gjEVM9odIA

I spent some time trying to tweak the RegExp to keep this fix and avoid the bug that introduced, but I have a feeling that tweaking the RegExp will not be enough, I think the `markInputRule` and `markPasteRule` functions play a part in this.

@marwan37 Think you can have a look since you were the original author of this change?

_(cc @nperez0111 for awareness; not sure if a fix for this is that straightforward, but if it's not, maybe a revert could be in order?)_

**Comment by rfgamaral** [comment]
> @rfgamaral what version are you using, because we've changed the regex since this was closed to remove the backtracking in: #5916

I tried with both the version that introduced this PR, and the latest, the issue exists on both.

That PR changed the implementation of the regex to not include lookbehind assertions (unsupported by Safari below 16.4), but it kept the RegExp change that introduces the new behaviour described in this PR (which also introduces the bug I just described).

**Pull request merged**